sentry-android-core/src/main/java/io/sentry/android/core/SentryPerformanceProvider.java

```diff
@@ -74,53 +55,145 @@ public String getType(@NotNull Uri uri) {
+    if (context instanceof Application) {
+      app = (Application) context;
+    }
+    if (app == null) {
+      return;
     }
-  }
 
-  @Override
-  public void onActivityPaused(@NotNull Activity activity) {}
+    final @NotNull TimeSpan appStartTimespan = appStartMetrics.getAppStartTimeSpan();
+    appStartTimespan.setStartedAt(Process.getStartUptimeMillis());
+
+    final AtomicBoolean firstDrawDone = new AtomicBoolean(false);
+    final Handler handler = new Handler(Looper.getMainLooper());
+
+    activityCallback =
+        new ActivityLifecycleCallbacksAdapter() {
+          final WeakHashMap<Activity, ActivityLifecycleTimeSpan> activityLifecycleMap =
+              new WeakHashMap<>();
+
+          @Override
+          public void onActivityPreCreated(
+              @NonNull Activity activity, @Nullable Bundle savedInstanceState) {
+            final long now = SystemClock.uptimeMillis();
+            if (appStartMetrics.getAppStartTimeSpan().hasStopped()) {
+              return;
+            }
+
+            final ActivityLifecycleTimeSpan timeSpan = new ActivityLifecycleTimeSpan();
+            timeSpan.onCreate.setStartedAt(now);
+            activityLifecycleMap.put(activity, timeSpan);
+          }
+
+          @Override
+          public void onActivityCreated(
+              @NonNull Activity activity, @Nullable Bundle savedInstanceState) {
+            if (appStartMetrics.getAppStartType() == AppStartMetrics.AppStartType.UNKNOWN) {
+              appStartMetrics.setAppStartType(
+                  savedInstanceState == null
+                      ? AppStartMetrics.AppStartType.COLD
+                      : AppStartMetrics.AppStartType.WARM);
+            }
+          }
+
+          @Override
+          public void onActivityPostCreated(
+              @NonNull Activity activity, @Nullable Bundle savedInstanceState) {
+            if (appStartMetrics.getAppStartTimeSpan().hasStopped()) {
+              return;
+            }
+
+            final @Nullable ActivityLifecycleTimeSpan timeSpan = activityLifecycleMap.get(activity);
+            if (timeSpan != null) {
+              timeSpan.onCreate.stop();
+              timeSpan.onCreate.setDescription(activity.getClass().getName() + ".onCreate");
+            }
+          }
+
+          @Override
+          public void onActivityPreStarted(@NonNull Activity activity) {
+            final long now = SystemClock.uptimeMillis();
+            if (appStartMetrics.getAppStartTimeSpan().hasStopped()) {
+              return;
+            }
+            final @Nullable ActivityLifecycleTimeSpan timeSpan = activityLifecycleMap.get(activity);
+            if (timeSpan != null) {
+              timeSpan.onStart.setStartedAt(now);
+            }
+          }
+
+          @Override
+          public void onActivityStarted(@NonNull Activity activity) {
+            if (firstDrawDone.get()) {
+              return;
+            }
+            NextDrawListener.forActivity(
```

Comment:
why not using `FirstDrawDoneListener.registerForNextDraw()`, as it's already there? Does it provide something more?